### DISCUSSION TITLE
【文章统计】不蒜子移动端和网页端次数不同

### DISCUSSION THREAD

**Discussion by tangly1024** [tangly1024/NotionNext#1951] [Q&A]
**描述bug**
来自讨论区
https://github.com/tangly1024/NotionNext/discussions/470
> 页面顶部的访问次数，移动端和桌面端差距很大，移动端刚发布的文章就300多次查看，应该不对，桌面端数据小很多。

**Comment by tangly1024** [comment]
https://www.v2ex.com/t/823654

**Comment by tangly1024** [comment]
https://github.com/ppoffice/hexo-theme-icarus/discussions/996

**Comment by tangly1024** [comment]
https://removeif.github.io/theme/不蒜子统计常见问题.html

**Comment by tangly1024** [comment]
解决方案 参考文章
- referer api
https://developer.mozilla.org/zh-CN/docs/Web/API/Document/referrer

- referer的作用 
https://www.ruanyifeng.com/blog/2019/06/http-referer.html

- referer不应该被修改
https://cloud.tencent.com/developer/article/1802164

- 伪造referer
https://www.cnblogs.com/zcm123/p/5977497.html

## 并非所有浏览器都尊重 refferer
https://jdhao.github.io/2020/10/31/busuanzi_pv_count_error/

https://github.com/next-theme/hexo-theme-next/issues/136

**Comment by expoli** [comment]
![图片](https://user-images.githubusercontent.com/31023767/213354658-5faed95d-d339-44b4-976e-68ba5b315a46.png)
文章界面回送的 referer 为域名信息，这应该是统计不准确的原因（浏览器：firefox）

**Comment by expoli** [comment]
![图片](https://user-images.githubusercontent.com/31023767/213355157-39b9e6f2-53f4-41d6-91e4-4a8645470f73.png)
edge 倒是可以

**Comment by expoli** [comment]
开源版“不蒜子”来了！这是一个基于 #Golang + #Redis 的简易访问量统计系统，你可以理解为自建不蒜子

🌐链接：https://github.com/soxft/busuanzi

可以考虑尝试一下这个

**Comment by expoli** [comment]
> 开源版“不蒜子”来了！这是一个基于 #Golang + #Redis 的简易访问量统计系统，你可以理解为自建不蒜子
> 
> 🌐链接：https://github.com/soxft/busuanzi
> 
> 可以考虑尝试一下这个

看介绍是对原来的不蒜子进行了优化，提升了浏览器兼容性

**Comment by tangly1024** [comment]
按照官方描述，确实做了兼容性的修复： https://github.com/soxft/busuanzi/wiki/usage


不过似乎要自建一个服务，并且使用Docker部署，上手就变得复杂了
